documentation/src/main/antora/modules/ROOT/pages/component-checkpoint.adoc

```diff
@@ -0,0 +1,279 @@
+The checkpointing frequency can be set to `RECORD` or `TIME`. The `RECORD` frequency saves the checkpoint after processing a certain number of records, while the `TIME` frequency saves the checkpoint after a certain time interval.
+
+When the component is restarted, the runtime calls the `start()` method. The connector then retrieves the checkpoint object and resumes processing from the last saved state from the configuration.
+
+When implementing checkpointing, it is recommended to use a separate configuration class for checkpointing. This class should be nested within the main configuration class and annotated with `@Checkpoint`. This ensures that the checkpointing configuration is separate from the main configuration and can be easily managed by the runtime.
+
+
+Sample configuration class with checkpointing annotations:
+[source,java]
+----
+    ...
+    @Checkpoint
+    @Version(value = 2, migrationHandler = CheckpointMigrationHandler.class)
+    public static class CheckPointInputConfig implements Serializable {
+
+        public enum Strategy {
+            BY_ID,
+            BY_DATE
+        }
+
+        @Option
+        @DefaultValue("BY_ID")
+        private Strategy strategy = Strategy.BY_ID;
+
+        @Option
+        private String startDate;
+
+        @Option
+        private int sinceId;
+
+        @Option
+        private String status;
+    }
+
+    @Data
+    @GridLayout(value = { @GridLayout.Row("user"), @GridLayout.Row("pass")})
+    @GridLayout(names = GridLayout.FormType.CHECKPOINT, value = { @GridLayout.Row("checkPointInputConfig") })
+    public static class InputConfig {
+
+        @Option
+        private String user;
+
+        @Option
+        private String pass;
+
+        @Option
+        private CheckPointInputConfig checkpoint = new CheckPointInputConfig();
+    }
+----
+
+Calling the `getCheckpoint()` returns `CheckpointState` class, which is a simple POJO class that holds the checkpoint data and version. This class has to be serializable to allow the runtime to save and restore the checkpoint object. For that, it provides a helper method `toJson()` to serialize the checkpoint state object.
+
+Here's a simple example of a `CheckpointState` class serialized to JSON:
+[source,json]
+----
+{
+  "$checkpoint": { <1>
+    "lastId": 95,
+    "lastUpdate": "2023-04-04",
+    "status": "running",
+    "strategy": "BY_ID",
+    "__version": 2 <2>
+  }
+}
+----
+Notice the following which are important (automatically done using `toJson()`):
+
+<1> Configuration and state of the checkpoint should be a member of a `$checkpoint` object.
+<2> Version of the checkpoint configuration is stored for eventual migration.
+
+The checkpointing configuration class is defined as a nested class within the component configuration class. The runtime has to provide the checkpointing configuration to the component when it is started.
+[source,java]
+----
+  ...
+  configuration.put("$checkpoint.strategy", "BY_ID");
```

Comment:
Don't remember well, but I think there were something magic to merge the connector's configuration withe the checkpoint one ?